### ISSUE TITLE
 Implementation

### ISSUE THREAD

**Issue by GlebSBrykin** [dypark86/SANET#1]
Hello! When will You open the source code?

**Comment by dypark86** [comment]
In current state, i have no plan to open source code.
I will open demo only at this site.

Because our company proceeding business with this algorithm.
so there are some problem in open source code.

instead, SANET consists of  'Adain' and 'Non local block'.
and these algorithm was published at github

so if you fully understanding paper.
you can implement SANET by modifying the below reference slightly. 
- Adain: https://github.com/naoto0804/pytorch-AdaIN 
- non local block self_attn class: https://github.com/heykeetae/Self-Attention-GAN/blob/master/sagan_models.py

Thanks!!

**Comment by GlebSBrykin** [comment]
Thanks for the reply. I've studied Your paper and the proposed papers, but I still have a question. Which is closer to SANet, to the Self-attention block you offered me, or to the non-local block, as the paper says? The only thing I would like to know is does convolution "h" change the depth(number of channels) of the input tensor, or not? This, unfortunately, is not written in the paper.

**Comment by GlebSBrykin** [comment]
Based on the fact that we perform a convolution before adding content to the Fcs, the input convolution "h" most likely changes the depth.

**Comment by dypark86** [comment]
I do not change the depth with convolution "h".
but if you want more light model, you can change the depth with convolution "h"

**Comment by GlebSBrykin** [comment]
Good. Then convolutions f and g must divide the number of channels to 8?

**Comment by dypark86** [comment]
no, i did not divide the f,h and g channels.

**Comment by GlebSBrykin** [comment]
Can I get more accurate information about the learning process: was learning rate decay and TVLoss used?

**Comment by dypark86** [comment]
I don't use TVLoss.
and other hyper parameter ref this Adain: https://github.com/naoto0804/pytorch-AdaIN

**Comment by GlebSBrykin** [comment]
Thanks for reply! I put the parameters from the link you provided, but the result was worse than it was before. Could you look at my code? Most likely, I made a mistake somewhere, and it is not in the parameters.
[sanet.zip](https://github.com/dypark86/SANET/files/3142274/sanet.zip)

**Comment by dypark86** [comment]
your identity 2 loss seems like incorrect.
identity 2 loss calculate mse loss with vgg feature 1 ~ 5(Icc vgg feature<-> content vgg feature, Iss vgg feature <-> style vgg feature).

**Comment by GlebSBrykin** [comment]
Thank you! But I rechecked my code several times and did not understand the error. In my case, calculate_content_loss calculates a normal mse loss if norm=False and it is false by default. Otherwise, the mse loss is calculated between the normalized input and the normalized target. Thus, content loss without parameters is a normal mse loss. If I understood correctly, identity2 loss is the same identity1 loss, only calculated not between images, but between vgg layers.

**Comment by GlebSBrykin** [comment]
Icc, as I understand it, is an image obtained by stylizing the content for itself. With Iss likewise, only for the style.

**Comment by GlebSBrykin** [comment]
I would still like to know what specific training parameters were used in the release model. Style weight, content weight, identity1 and identity2 weight, learning rate and lr_decay. From Your paper, for example, it is not clear, the weight of the style you have 3 or 5. Regarding the rest of the same question. If do so, as You advised me in the preceding summit(all parameters simply copy with AdaIN), then the result is highly strange. It has nothing to do with what was given in the paper.

**Comment by GlebSBrykin** [comment]
Actually, if I take parameters from your paper and train them for 50,000 iterations, I get this result:
[test, sw=3, cw=1, i1=1, i2=50, lr=1e-4, lr_d=5e-5, 50000 it.zip](https://github.com/dypark86/SANET/files/3151997/test.sw.3.cw.1.i1.1.i2.50.lr.1e-4.lr_d.5e-5.50000.it.zip)

**Comment by dypark86** [comment]
there are some mistake in hyper parameter

use identity1 = 50 identity 2 = 1

and  use 3*3conv 3layer without relu, after style attention feature sum.
